File: Medium/0012-Integer to Roman/Solution.py
class Solution:
    def intToRoman(self, num: int) -> str:
        roman = (num // 1000) * 'M'
        roman += (num % 1000 // 500) * 'D'
        roman += (num % 500 // 100) * 'C'
        roman += (num % 100 // 50) * 'L'
        roman += (num % 50 // 10) * 'X'
        roman += (num % 10 // 5) * 'V'
        roman += (num % 5) * 'I'

        roman = roman.replace("DCCCC", "CM")
        roman = roman.replace("CCCC", "CD")
        roman = roman.replace("LXXXX", "XC")
        roman = roman.replace("XXXX", "XL")
        roman = roman.replace("VIIII", "IX")
        roman = roman.replace("IIII", "IV")

        return roman

File: Medium/0012-Integer to Roman/Solution.rb
# @param {Integer} num
# @return {String}
def int_to_roman(num)
    roman = 'M' * (num / 1000)
    roman += 'D' * (num % 1000 / 500)
    roman += 'C' * (num % 500 / 100)
    roman += 'L' * (num % 100 / 50)
    roman += 'X' * (num % 50 / 10)
    roman += 'V' * (num % 10 / 5)
    roman += 'I' * (num % 5)

    roman.sub!("DCCCC", "CM")
    roman.sub!("CCCC", "CD")
    roman.sub!("LXXXX", "XC")
    roman.sub!("XXXX", "XL")
    roman.sub!("VIIII", "IX")
    roman.sub!("IIII", "IV")

    return roman
end
